scvi/inference/posterior.py

```diff
@@ -1269,13 +1304,25 @@ def get_sample_scale(self, transform_batch=None):
                             sample_batch,
                             batch_index=batch_index,
                             y=labels,
-                            n_samples=1,
+                            n_samples=n_samples,
                             transform_batch=transform_batch,
-                        )
+                        )[..., gene_mask] * library_size
                     ).cpu()
                 )
             ]
-        return np.concatenate(px_scales)
+
+        if n_samples > 1:
+            px_scales = np.concatenate(px_scales, axis=-2)
```

Comment:
Can't this one and the one below both be axis 0? If `n_samples > 1`, then isn't the shape `(samples, cells, genes)`?

Reply:
ah, no sorry -2 is cells, at least a comment would help us remember :)